Drop operators that are no longer necessary with C++20

cdumez opened the pull request:
#### 74d33a84ca597da6e2be2416b3586468538cdc0c
<pre>
Drop operators that are no longer necessary with C++20
<a href="https://bugs.webkit.org/show_bug.cgi?id=291446">https://bugs.webkit.org/show_bug.cgi?id=291446</a>

Reviewed by Michael Catanzaro.

Drop operators that are no longer necessary with C++20:
- `operator==(b, a)` is no longer needed when `operator==(a, b)` exists
- `operator!=(a, a)` is no longer needed when operator==(a, a)` exists

* Source/JavaScriptCore/API/JSRetainPtr.h:
(operator==):
* Source/JavaScriptCore/dfg/DFGEdge.h:
(JSC::DFG::operator==):
* Source/JavaScriptCore/heap/Handle.h:
* Source/JavaScriptCore/heap/WeakInlines.h:
* Source/JavaScriptCore/runtime/JSCJSValue.h:
(JSC::operator==):
* Source/JavaScriptCore/runtime/PropertyName.h:
* Source/WTF/wtf/CompactPtr.h:
* Source/WTF/wtf/Expected.h:
(std::experimental::fundamentals_v3::operator==):
* Source/WTF/wtf/Float16.h:
(WTF::Float16::operator!= const): Deleted.
* Source/WTF/wtf/Markable.h:
(WTF::operator==):
* Source/WTF/wtf/Packed.h:
* Source/WTF/wtf/RefPtr.h:
* Source/WTF/wtf/RetainPtr.h:
* Source/WTF/wtf/SortedArrayMap.h:
* Source/WTF/wtf/URL.h:
* Source/WTF/wtf/WeakPtr.h:
* Source/WTF/wtf/cocoa/Entitlements.mm:
(WTF::hasEntitlementValueInArray):
* Source/WTF/wtf/glib/GRefPtr.h:
(WTF::operator!=): Deleted.
* Source/WTF/wtf/text/AtomString.h:
(WTF::operator==):
* Source/WTF/wtf/text/StringView.h:
(WTF::operator==):
* Source/WTF/wtf/text/WTFString.h:
(WTF::operator==):
* Source/WebCore/css/CSSSelector.h:
* Source/WebCore/loader/cache/CachedResourceHandle.h:
(WebCore::operator==):
* Source/WebCore/platform/encryptedmedia/CDMProxy.h:
(WebCore::KeyHandle::operator==):
* Source/WebCore/platform/graphics/gstreamer/GStreamerCommon.h:
(WebCore::unmapFunction):
* Source/WebCore/platform/win/COMPtr.h:
(operator!=): Deleted.
* Source/WebKit/UIProcess/API/cpp/WKRetainPtr.h:

Canonical link: <a href="https://commits.webkit.org/293614@main">https://commits.webkit.org/293614@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/239c1153d6e179653314402111ba444bd0fc5a65

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/99386 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/131/builds/19035 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/9287 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/104517 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/49987 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/101427 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/130/builds/19324 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/27469 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/75647 "Passed tests") | [✅ 🧪 win-tests](https://ews-build.webkit.org/#/builders/60/builds/32744 "Passed tests") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/102393 "Passed tests") | [❌ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/132/builds/14698 "Found 1 new test failure: fast/forms/ios/file-upload-panel-dismiss-when-view-removed-from-window.html (failure)") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/89732 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/56006 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/133/builds/14492 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/135/builds/7715 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/49347 "Built successfully") | 
| [✅ 🛠 🧪 jsc](https://ews-build.webkit.org/#/builders/20/builds/92069 "Built successfully and passed tests") | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/84418 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/136/builds/7802 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/106874 "Built successfully") | 
| [✅ 🛠 🧪 jsc-arm64](https://ews-build.webkit.org/#/builders/12/builds/98005 "Built successfully and passed tests") | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/26500 "Built successfully") | [  ~~🧪 mac-AS-debug-wk2~~](https://ews-build.webkit.org/#/builders/122/builds/19333 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [❌ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/84609 "Found 1 new test failure: imported/w3c/web-platform-tests/service-workers/service-worker/redirected-response.https.html (failure)") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/26862 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/85936 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/84122 "Passed tests") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/21342 "Built successfully and passed tests") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/28787 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/137/builds/6476 "Passed tests") | [✅ 🛠 playstation](https://ews-build.webkit.org/#/builders/134/builds/20243 "Built successfully") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/26440 "Built successfully") | [❌ 🛠 mac-safer-cpp](https://ews-build.webkit.org/#/builders/120/builds/31641 "Found 1 new failure in runtime/BrandedStructure.h and found 1 fixed file: wtf/cocoa/Entitlements.mm") | [✅ 🛠 jsc-armv7](https://ews-build.webkit.org/#/builders/35/builds/121621 "Built successfully") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/26260 "Built successfully") | | [✅ 🧪 jsc-armv7-tests](https://ews-build.webkit.org/#/builders/25/builds/33969 "Passed tests") | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/29573 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/27827 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

mcatanzaro commented on Source/WTF/wtf/cocoa/Entitlements.mm:
```diff
@@ -88,10 +88,10 @@ bool hasEntitlementValueInArray(audit_token_t token, ASCIILiteral entitlement, A
-        auto element = CFArrayGetValueAtIndex(array.get(), i);
-        if (CFGetTypeID(element) != CFStringGetTypeID())
+        RetainPtr element = dynamic_cf_cast<CFStringRef>(CFArrayGetValueAtIndex(array.get(), i));
+        if (!element)
             continue;
-        CFStringRef stringElement = static_cast<CFStringRef>(element);
+        String stringElement = element.get();
```
Looks like this was intended for a different pull request.

cdumez replied:
No this is intentional. We were comparing a CFStringRef and an ASCIILiteral and the compiler needed some help to call the right operator.

cdumez committed:
Drop operators that are no longer necessary with C++20
https://bugs.webkit.org/show_bug.cgi?id=291446

Reviewed by Michael Catanzaro.

Drop operators that are no longer necessary with C++20:
- `operator==(b, a)` is no longer needed when `operator==(a, b)` exists
- `operator!=(a, a)` is no longer needed when operator==(a, a)` exists

* Source/JavaScriptCore/API/JSRetainPtr.h:
(operator==):
* Source/JavaScriptCore/dfg/DFGEdge.h:
(JSC::DFG::operator==):
* Source/JavaScriptCore/heap/Handle.h:
* Source/JavaScriptCore/heap/WeakInlines.h:
* Source/JavaScriptCore/runtime/JSCJSValue.h:
(JSC::operator==):
* Source/JavaScriptCore/runtime/PropertyName.h:
* Source/WTF/wtf/CompactPtr.h:
* Source/WTF/wtf/Expected.h:
(std::experimental::fundamentals_v3::operator==):
* Source/WTF/wtf/Float16.h:
(WTF::Float16::operator!= const): Deleted.
* Source/WTF/wtf/Markable.h:
(WTF::operator==):
* Source/WTF/wtf/Packed.h:
* Source/WTF/wtf/RefPtr.h:
* Source/WTF/wtf/RetainPtr.h:
* Source/WTF/wtf/SortedArrayMap.h:
* Source/WTF/wtf/URL.h:
* Source/WTF/wtf/WeakPtr.h:
* Source/WTF/wtf/cocoa/Entitlements.mm:
(WTF::hasEntitlementValueInArray):
* Source/WTF/wtf/glib/GRefPtr.h:
(WTF::operator!=): Deleted.
* Source/WTF/wtf/text/AtomString.h:
(WTF::operator==):
* Source/WTF/wtf/text/StringView.h:
(WTF::operator==):
* Source/WTF/wtf/text/WTFString.h:
(WTF::operator==):
* Source/WebCore/css/CSSSelector.h:
* Source/WebCore/loader/cache/CachedResourceHandle.h:
(WebCore::operator==):
* Source/WebCore/platform/encryptedmedia/CDMProxy.h:
(WebCore::KeyHandle::operator==):
* Source/WebCore/platform/graphics/gstreamer/GStreamerCommon.h:
(WebCore::unmapFunction):
* Source/WebCore/platform/win/COMPtr.h:
(operator!=): Deleted.
* Source/WebKit/UIProcess/API/cpp/WKRetainPtr.h:

Canonical link: https://commits.webkit.org/293614@main
